CMakeLists.txt

```diff
@@ -210,112 +205,161 @@ target_compile_definitions(vcpkglib PUBLIC
+    )
 
-        add_custom_target(rush-update
-            ALL
-            DEPENDS
-                "${CMAKE_CURRENT_SOURCE_DIR}/ce/ce/node_modules"
-                "${CMAKE_CURRENT_SOURCE_DIR}/ce/test/node_modules"
-                )
-
-# === Target: vcpkg-artifacts ===
-        file(GLOB_RECURSE VCPKG_ARTIFACTS_SOURCES CONFIGURE_DEPENDS "ce/ce/*.ts")
-        add_custom_command(
-            OUTPUT
-                "${CMAKE_CURRENT_LIST_DIR}/ce/ce/dist/tsconfig.tsbuildinfo"
-            COMMAND
-                "${NODEJS}" "${CMAKE_CURRENT_LIST_DIR}/ce/ce/node_modules/typescript/bin/tsc"
-                    -p "${CMAKE_CURRENT_LIST_DIR}/ce/ce"
-            DEPENDS
-                ${VCPKG_ARTIFACTS_SOURCES}
-                "${CMAKE_CURRENT_SOURCE_DIR}/ce/ce/node_modules"
-            COMMENT
-                "Building vcpkg-artifacts..."
-            VERBATIM
-            )
+# === Target: vcpkg-artifacts-target ===
+# The suffix "-target" is added to avoid a conflict in CMake with the directory named vcpkg-artifacts and the target named vcpkg-artifacts.
+    file(GLOB VCPKG_ARTIFACTS_ROOT_SOURCES LIST_DIRECTORIES false RELATIVE "${VCPKG_ARTIFACTS_SOURCE_DIR}" CONFIGURE_DEPENDS "${VCPKG_ARTIFACTS_SOURCE_DIR}/*.ts")
+    file(GLOB_RECURSE VCPKG_ARTIFACTS_NESTED_SOURCES LIST_DIRECTORIES false RELATIVE "${VCPKG_ARTIFACTS_SOURCE_DIR}" CONFIGURE_DEPENDS
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/amf/*.ts"
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/archivers/*.ts"
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/artifacts/*.ts"
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/cli/*.ts"
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/fs/*.ts"
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/installers/*.ts"
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/interfaces/*.ts"
+        # "${VCPKG_ARTIFACTS_SOURCE_DIR}/locales/*.ts" 
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/mediaquery/*.ts"
+        # "${VCPKG_ARTIFACTS_SOURCE_DIR}/node_modules/*.ts"
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/registries/*.ts"
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/test/*.ts"
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/test-resources/*" # Note no *.ts
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/util/*.ts"
+        "${VCPKG_ARTIFACTS_SOURCE_DIR}/yaml/*.ts"
+    )
+    set(VCPKG_ARTIFACTS_SOURCES ${VCPKG_ARTIFACTS_ROOT_SOURCES} ${VCPKG_ARTIFACTS_NESTED_SOURCES})
+    list(TRANSFORM VCPKG_ARTIFACTS_SOURCES PREPEND "${VCPKG_ARTIFACTS_SOURCE_DIR}/")
 
-        add_custom_target(vcpkg-artifacts
-            ALL
-            DEPENDS
-                "${CMAKE_CURRENT_LIST_DIR}/ce/ce/dist/tsconfig.tsbuildinfo"
-            )
+    set(VCPKG_ARTIFACTS_BINARIES ${VCPKG_ARTIFACTS_ROOT_SOURCES} ${VCPKG_ARTIFACTS_NESTED_SOURCES}) 
+    list(TRANSFORM VCPKG_ARTIFACTS_BINARIES PREPEND "${VCPKG_ARTIFACTS_BINARY_DIR}/")
+    list(TRANSFORM VCPKG_ARTIFACTS_BINARIES REPLACE "\\.ts" ".js")
+    add_custom_command(
+        OUTPUT "${VCPKG_ARTIFACTS_SOURCE_DIR}/locales/messages.json"
```

Comment:
Currently, yes. There are probably ways to avoid it but I'm changing enough about the status quo here.

I believe the main reason there is this trip through the source directory is to not disrupt contributors who aren't touching the artifacts part when `VCPKG_ARTIFACTS_DEVELOPMENT` is turned off.

@JavierMatosD Do you remember any other rationale for putting this file in the source tree?